cmd/root.go

```diff
@@ -168,6 +135,8 @@ func initViperConfig() {
 		if config.CfgFile != "" || !cfgNotFound {
 			cobra.CheckErr(err)
 		}
+	} else {
+		diagnostics.ConfigFileLoaded = true
```

Comment:
I remember going back & forth about whether to do that. I think we'll add more diagnostics over time, this one was kinda obvious and yeah, we should grab any error off that Unmarshal and check it and put it in diagnostics too.

I'm kinda tempted to drop Viper and config files all together and just do envvars manually. It would drop a lot of transitive dependencies and fewer inputs might make it easier to maintain...

Reply:
im down for lower dependency count. I would like to try out [koanf](https://github.com/knadh/koanf) and maybe even a full manual approach as an optimization in a separate issue and workstream.